authz/api/__init__.py

```diff
@@ -1,77 +1,143 @@
-from flask import Blueprint, jsonify, abort, request
-from sqlalchemy.orm.exc import NoResultFound
+from flask import Blueprint, jsonify, abort, request, Response
 
-from authz.models import Role, User, WorkspaceRole
 from authz.serializers.role import RoleSerializer
+from authz.serializers.user import UserSerializer
+from authz.serializers.workspace_user import WorkspaceUserSerializer
 
-api = Blueprint('api', __name__)
+from authz.domain.exceptions import NotFoundError, AlreadyExistsError
+from authz.domain.workspace_users import WorkspaceUsers
+from authz.domain.users import Users
+from authz.domain.roles import Roles
 
+api = Blueprint("api", __name__)
 
-@api.route('/roles')
+
+@api.route("/roles")
 def get_roles():
-    roles = Role.query.all()
+    """
+    Returns a list of all existing roles.
+
+    GET /roles
+    """
+
+    roles = Roles.get_all()
     return RoleSerializer().jsonify(roles, many=True)
 
 
-@api.route('/roles/<string:name>')
+@api.route("/roles/<string:name>")
 def get_role(name):
+    """
+    Returns a given role.
+
+    GET /roles/<role name>
+    """
+
     try:
-        role = Role.query.filter_by(name=name).one()
-    except NoResultFound:
-        return abort(404)
+        role = Roles.get(name)
+    except NotFoundError as e:
+        return (jsonify({"error": e.message}), 404)
+
     return RoleSerializer().jsonify(role)
 
 
-@api.route('/workspaces/<uuid:workspace_id>/users', methods=['PUT'])
+@api.route("/users", methods=["POST"])
+def create_user():
+    """
+    Creates a new atat user with the given role.
+
+    POST /users/
+    {
+        id: <user id>,
```

Comment:
Do we want the users of the API to define the id, or should this service handle creating the id?

Reply:
The id for users in this case is the same as the one in `authnid`, which is the service that "owns" users. We could have a separate `authnid_id` (or better named) param and column instead if you think that makes more sense.